### ISSUE TITLE
Feature request: reload data files once changes are detected

### ISSUE THREAD

**Issue by randomizedthinking** [lfos/calcurse#45]
I want to use calcurse at different machines, and have set up auto-sync for all files under `.calcurse`. Yet, what I found is that calcurse won't check and load the updated data files; hence, data constantly get loss if I have existing calcurse sessions at other machines.

Here, I would like to request the feature for calcurse to auto-reload data and configuration files once changes are detected just as vim does. Regards.

**Comment by lfos** [comment]
I pushed a partial solution to the `pu` branch. With d483407 (Do not blindly overwrite files when saving, 2017-09-07), calcurse checks whether the data files were changed before saving. If they were, you can either overwrite, cancel or run the merge tool to keep both changes.

**Comment by lfos** [comment]
I will look into implementing an auto-reload feature as soon as possible.

**Comment by randomizedthinking** [comment]
Thanks a bunch!

**Comment by lfos** [comment]
I looked into it but it is a bit more involved than I thought. For example, it is not clear what to do if an appointment is deleted in the data files and the reload is triggered while the user is currently editing that appointment...

**Comment by randomizedthinking** [comment]
Yes, it is involved and the implementation can be simple to cover most cases. Both the save and reload functions should rely on the timestamp of the data files. For the scenario you described, the editing should be discarded. A prompt can be provided or skipped based on user configueration.

Just my 2 cents.

**Comment by lfos** [comment]
Discarding the editing is not a good idea because the user will be surprised. I think we should rather turn off the automatic reload temporarily during any interactive edit operations...

**Comment by mvhulten** [comment]
I agree with @lfos: in normal operation check for and update changes.  When editing a new appointment, todo or note, don't do that.

But then you don't cover some common synchronisation issues, which you appear to cover with your patch.

Does CalDAV accomplish the same?  Or does that have disadvantages, e.g. overkill, over a solution within calcurse?

**Comment by lfos** [comment]
I don't quite understand. How is CalDAV an alternative to reloading files once changes are detected?

**Comment by mvhulten** [comment]
@randomizedthinking wanted to use calcurse at different machines and be sync'ed between them.  Would CalDAV not do that (with the need of a CalDAV server, of course)?

Though even if one would accept this as a solution for synchronising between different calcurse sessions, there may be additional reasons for wanting to check for changes.

**Comment by mvhulten** [comment]
@randomizedthinking wanted to use calcurse at different machines and be sync'ed between them.  Would CalDAV not do that (with the need of a CalDAV server, of course)?

Though even if one would accept this as a solution for synchronising between different calcurse sessions, there may be additional reasons for wanting to check for changes.
